sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1300,354 +1302,360 @@ def wait_for_port_config_done(self, namespace):
             if key in ["PortConfigDone", "PortInitDone"]:
                 break
 
-    def task_worker(self):
-        self.xcvr_table_helper = XcvrTableHelper(self.namespaces)
-
-        self.log_notice("Waiting for PortConfigDone...")
-        for namespace in self.namespaces:
-            self.wait_for_port_config_done(namespace)
-
-        logical_port_list = self.port_mapping.logical_port_list
-        for lport in logical_port_list:
-            self.update_port_transceiver_status_table_sw_cmis_state(lport, CMIS_STATE_UNKNOWN)
-
-        is_fast_reboot = is_fast_reboot_enabled()
-
-        # APPL_DB for CONFIG updates, and STATE_DB for insertion/removal
-        port_change_observer = PortChangeObserver(self.namespaces, helper_logger,
-                                                  self.task_stopping_event,
-                                                  self.on_port_update_event)
-
-        while not self.task_stopping_event.is_set():
-            # Handle port change event from main thread
-            port_change_observer.handle_port_update_event()
-
-            for lport, info in self.port_dict.items():
-                if self.task_stopping_event.is_set():
-                    break
-
-                if lport not in self.port_dict:
-                    continue
-
-                state = get_cmis_state_from_state_db(lport, self.xcvr_table_helper.get_status_tbl(self.port_mapping.get_asic_id_for_logical_port(lport)))
-                if state in CMIS_TERMINAL_STATES or state == CMIS_STATE_UNKNOWN:
-                    if state != CMIS_STATE_READY:
-                        self.port_dict[lport]['appl'] = 0
-                        self.port_dict[lport]['host_lanes_mask'] = 0
-                    continue
+    def handle_cmis_state_machine(self, lport, info, is_fast_reboot):
```

Comment:
The change here seems to be refactoring today's cmis_mgr's `task_worker` function by splitting it into sub functions. I'm assuming the flow/logic of cmis_mgr's task_worker doesn't get changed, it's mostly just splitting functions. But this makes the number of LOC (Line of change) very big. I feel It's a little bit difficult for other people to distinguish the actual change of DPB fix from the entire change set of this PR.

Thus, I feel it might be better to put this refactoring change set in a separate PR. It will also be less of a work for release mgr to revert in case of anything broken. Just my 2 cents, not something must-have.
Not sure how @mihirpat1 or Prince feel about this.

Reply:
@longhuan-cisco Agreed that splitting the current changeset into 2 separate PRs will be more helpful.

Reply:
@longhuan-cisco @mihirpat1 I'm sorry for the late response. Thanks for the suggestion.

I'll move the refactoring commit to a separate PR and only include the DPB fix in this PR.